### PR TITLE
REL-2030: Invalid character causes proposal to be rejected

### PR DESCRIPTION
**Pull request by cquiroz**
Quick PR, sometimes people edit their proposals by hand and break the XML, this PR takes care of filtering out non valid xml files

### CODE REVIEW

File: bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/ProposalIo.scala [hunk 1]
```diff
@@ -50,15 +50,26 @@ object ProposalIo {
...
-  //private def tryLatin1(f: File)
-
   private def toProposal(r: ConversionResult): ProposalConversion = ProposalConversion(r.transformed, r.from, r.changes, toProposal(unmarshaller.unmarshal(new StringReader(r.root.toString))))
 
-  def readAndConvert(s: String):Validation[Either[Exception, NonEmptyList[String]], ProposalConversion] = try {
-      UpConverter.upConvert(XML.loadString(s)).fold(Right(_).fail, toProposal(_).success)
+  // Not all UFT-8 characters are valid in XML, strip them if needed
+  // Reference http://www.w3.org/TR/2000/REC-xml-20001006#NT-Char
+  private def stripNonValidXMLCharacters(in: String) = in.filter {
+    case c if c == 0x9                      => true
```
Comment (sraaphorst) [inline bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/ProposalIo.scala]: I'm not sure why these single cases have the form:
`case c if c == 0x9 => true`
Wouldn't it be simpler to do:
`case 0x9 => true`?

(Or even: `case c if c in List(0x9, 0xA, 0xD) => true` and just combine them?)

Comment (tpolecat) [inline bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/ProposalIo.scala]:
```scala
case 0x9 | 0xA | 0xD => true
```

Comment (sraaphorst) [inline bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/ProposalIo.scala]: Yup. That makes more sense. I always forget about that syntax.

Comment (cquiroz) [inline bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/ProposalIo.scala]: Thanks for the suggestions, I updated the code acoordingly

**Comment by swalker2m** [comment]
:+1:

**Pull request merged**